Operator is panicking when pvc size is updated externally

**Describe the bug**

Operator is panicking when PVC size is updated
```text
{"level":"info","ts":1592473490.1457653,"logger":"controllers.KafkaCluster","msg":"Kafka cluster state updated","Request.Namespace":"ns-team-aep-pipeline-kafka-1-dev/pipeline-kafka","Request.Name":"pipeline-kafka","component":"kafka","clusterName":"pipeline-kafka","clusterNamespace":"ns-team-aep-pipeline-kafka-1-dev"}
E0618 09:44:50.248089       1 runtime.go:78] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
goroutine 541 [running]:
k8s.io/apimachinery/pkg/util/runtime.logPanic(0x1928340, 0x2cf5880)
	/go/pkg/mod/k8s.io/apimachinery@v0.18.2/pkg/util/runtime/runtime.go:74 +0xa3
k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	/go/pkg/mod/k8s.io/apimachinery@v0.18.2/pkg/util/runtime/runtime.go:48 +0x82
panic(0x1928340, 0x2cf5880)
	/usr/local/go/src/runtime/panic.go:679 +0x1b2
github.com/banzaicloud/kafka-operator/pkg/errorfactory.(*InternalError).Error(0xc000ef17b0, 0x1912780, 0x19a2b20)
	<autogenerated>:1 +0x2e
github.com/pkg/errors.(*withMessage).Error(0xc001152be0, 0x0, 0xc001085920)
	/go/pkg/mod/github.com/pkg/errors@v0.9.0/errors.go:244 +0x37
emperror.dev/errors.(*withDetails).Error(0xc00108acf0, 0xc000ef17b0, 0x1e447a0)
	/go/pkg/mod/emperror.dev/errors@v0.7.0/error_details.go:62 +0x33
github.com/banzaicloud/kafka-operator/controllers.(*KafkaClusterReconciler).Reconcile(0xc00061a9b0, 0xc00041d3c0, 0x20, 0xc000a5b4a0, 0xe, 0xc000e2fca0, 0xc000305200, 0xc0003042d8, 0x1e5af40)
	/workspace/controllers/kafkacluster_controller.go:158 +0xca6
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler(0xc0001d2300, 0x19a8fc0, 0xc0007704e0, 0x0)
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.6.0/pkg/internal/controller/controller.go:256 +0x162
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc0001d2300, 0x40a700)
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.6.0/pkg/internal/controller/controller.go:232 +0xcb
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).worker(0xc0001d2300)
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.6.0/pkg/internal/controller/controller.go:211 +0x2b
k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0xc000bcb2e0)
	/go/pkg/mod/k8s.io/apimachinery@v0.18.2/pkg/util/wait/wait.go:155 +0x5e
k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0xc000bcb2e0, 0x1e434a0, 0xc000bc2f00, 0xc000a92901, 0xc00034a2a0)
	/go/pkg/mod/k8s.io/apimachinery@v0.18.2/pkg/util/wait/wait.go:156 +0xa3
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc000bcb2e0, 0x3b9aca00, 0x0, 0x1, 0xc00034a2a0)
	/go/pkg/mod/k8s.io/apimachinery@v0.18.2/pkg/util/wait/wait.go:133 +0xe2
k8s.io/apimachinery/pkg/util/wait.Until(0xc000bcb2e0, 0x3b9aca00, 0xc00034a2a0)
	/go/pkg/mod/k8s.io/apimachinery@v0.18.2/pkg/util/wait/wait.go:90 +0x4d
created by sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.6.0/pkg/internal/controller/controller.go:193 +0x328
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x137507e]
goroutine 541 [running]:
k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	/go/pkg/mod/k8s.io/apimachinery@v0.18.2/pkg/util/runtime/runtime.go:55 +0x105
panic(0x1928340, 0x2cf5880)
	/usr/local/go/src/runtime/panic.go:679 +0x1b2
github.com/banzaicloud/kafka-operator/pkg/errorfactory.(*InternalError).Error(0xc000ef17b0, 0x1912780, 0x19a2b20)
	<autogenerated>:1 +0x2e
github.com/pkg/errors.(*withMessage).Error(0xc001152be0, 0x0, 0xc001085920)
	/go/pkg/mod/github.com/pkg/errors@v0.9.0/errors.go:244 +0x37
emperror.dev/errors.(*withDetails).Error(0xc00108acf0, 0xc000ef17b0, 0x1e447a0)
	/go/pkg/mod/emperror.dev/errors@v0.7.0/error_details.go:62 +0x33
github.com/banzaicloud/kafka-operator/controllers.(*KafkaClusterReconciler).Reconcile(0xc00061a9b0, 0xc00041d3c0, 0x20, 0xc000a5b4a0, 0xe, 0xc000e2fca0, 0xc000305200, 0xc0003042d8, 0x1e5af40)
	/workspace/controllers/kafkacluster_controller.go:158 +0xca6
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler(0xc0001d2300, 0x19a8fc0, 0xc0007704e0, 0x0)
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.6.0/pkg/internal/controller/controller.go:256 +0x162
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc0001d2300, 0x40a700)
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.6.0/pkg/internal/controller/controller.go:232 +0xcb
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).worker(0xc0001d2300)
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.6.0/pkg/internal/controller/controller.go:211 +0x2b
k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0xc000bcb2e0)
	/go/pkg/mod/k8s.io/apimachinery@v0.18.2/pkg/util/wait/wait.go:155 +0x5e
k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0xc000bcb2e0, 0x1e434a0, 0xc000bc2f00, 0xc000a92901, 0xc00034a2a0)
	/go/pkg/mod/k8s.io/apimachinery@v0.18.2/pkg/util/wait/wait.go:156 +0xa3
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc000bcb2e0, 0x3b9aca00, 0x0, 0x1, 0xc00034a2a0)
	/go/pkg/mod/k8s.io/apimachinery@v0.18.2/pkg/util/wait/wait.go:133 +0xe2
k8s.io/apimachinery/pkg/util/wait.Until(0xc000bcb2e0, 0x3b9aca00, 0xc00034a2a0)
	/go/pkg/mod/k8s.io/apimachinery@v0.18.2/pkg/util/wait/wait.go:90 +0x4d
created by sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.6.0/pkg/internal/controller/controller.go:193 +0x328
```
**Steps to reproduce the issue:**
- Edit one of the broker PVC request size from X to X+1 using `kubectl edit pvc ...`
- kafka-operator spits the error above

**Expected behavior**
- operator should handle this gracefully and print and error

--- a/pkg/resources/kafka/kafka.go
+++ b/pkg/resources/kafka/kafka.go
@@ -804,7 +804,8 @@ func (r *Reconciler) reconcileKafkaPvc(log logr.Logger, brokersDesiredPvcs map[s
 					}
 
 					if isDesiredStorageValueInvalid(desiredPvc, currentPvc) {
-						return errorfactory.New(errorfactory.InternalError{}, nil, "one can not reduce the size of a PVC", "kind", desiredType)
+						return errorfactory.New(errorfactory.InternalError{}, errors.New("could not modify pvc size"),
+							"one can not reduce the size of a PVC", "kind", desiredType)
 					}
 
 					resReq := desiredPvc.Spec.Resources.Requests

